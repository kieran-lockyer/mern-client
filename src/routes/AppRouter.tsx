import React, { Component } from "react";
import Dashboard from "../components/Dashboard/Dashboard";
import Sidebar from "../components/Sidebar";
import PhotoList from "../components/Photos/PhotoList";
import PhotoSingle from "../components/Photos/PhotoSingle";
import TagList from "../components/Tags/TagList";
import TagSingle from "../components/Tags/TagSingle";
import NotFound from "../components/NotFound";
import { Router, Route, Switch, Redirect } from "react-router-dom";
import history from "../history";
import styled from "styled-components";

export default class App extends Component {
  render() {
    return (
      <Router history={history}>
        <Container>
          <Sidebar />
          <Switch>
            <Route path="/" exact component={Dashboard} />
            <Route
              path="/photos"
              exact
              render={() => <Redirect from="/photos" exact to="/photos/1" />}
            />
            <Route path="/photos/:id" exact component={PhotoList} />
            <Route path="/photo/:photoId" exact component={PhotoSingle} />
            <Route
              path="/tags"
              exact
              render={() => <Redirect from="/tags" exact to="/tags/1" />}
            />
            <Route path="/tags/:id" exact component={TagList} />
            <Route path="/tag/:tagId" exact component={TagSingle} />
            <Route component={NotFound} />
          </Switch>
        </Container>
      </Router>
    );
  }
}

const Container = styled.div`
  min-height: 100vh;
  width: 100%;
  display: flex;
  background: #eee;
  @media only screen and (max-width: 800px) {
    display: block;
  }
`;
