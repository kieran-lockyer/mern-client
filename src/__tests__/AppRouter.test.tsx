import React from 'react';
import { shallow } from 'enzyme';
import AppRouter from '../routes/AppRouter';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('<AppRouter />', () => {
  describe('render()', () => {
    test('renders the AppRouter component', () => {
      const wrapper = shallow(<AppRouter />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});