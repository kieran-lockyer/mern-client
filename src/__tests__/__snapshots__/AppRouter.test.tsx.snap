// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRouter /> render() renders the AppRouter component 1`] = `
<styled.div>
  <Sidebar />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      exact={true}
      path="/photos"
      render={[Function]}
    />
    <Route
      component={[Function]}
      exact={true}
      path="/photos/:id"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/photo/:photoId"
    />
    <Route
      exact={true}
      path="/tags"
      render={[Function]}
    />
    <Route
      component={[Function]}
      exact={true}
      path="/tags/:id"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/tag/:tagId"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</styled.div>
`;
